frontend/model/contracts/group.js

```diff
@@ -521,6 +525,27 @@ sbp('chelonia/defineContract', {
         }
       }
     },
+    'gi.contracts/group/sendPaymentThankYou': {
+      validate: objectOf({
+        fromUser: string,
+        toUser: string,
+        memo: string
+      }),
+      process ({ data, meta, hash }, { state }) {
+        Vue.set(state.paymentThankYouNotes, hash, data)
```

Comment:
By creating a `paymentThankYouNotes` this way - will save and store all payment thank yous forever. That seems excessive to me.

Instead of that, how about we just save the most recent thank-you for each user?

You can do this by creating a 2-dimentional lookup object, similar to how `paymentsFrom` works. You could call it, `thankYousFrom`, and access it like this:

```js
state.thankYousFrom[fromUser]?.[toUser]
```

I recommend also using `vueFetchInitKV` when implementing this. Have a look for example at how the process function for `'gi.contracts/group/payment'` uses it.

Then, for the `linkTo` in the notification, pass in the `fromUser` and `toUser` parameters to the modal.

Also, remember - if you are modifying contracts you need to pin a new version. For now it's OK that you haven't, because @Silver-IT is also modifying them in his PR, and his PR might be merged before yours. So his might be `0.0.5`, and yours might be `0.0.6`.